src/components/SurveyCourseList.tsx

```diff
@@ -0,0 +1,173 @@
+import { gql, useQuery } from "@apollo/client";
+import {
+  Divider,
+  Flex,
+  FormLabel,
+  Grid,
+  GridItem,
+  Heading,
+  Radio,
+  RadioGroup,
+  Stack,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { CourseInterface } from "../pages/Survey";
+
+const COURSES = gql`
+  query GetCourses {
+    survey {
+      courses {
+        subject
+        code
+        term
+      }
+    }
+  }
+`;
+
+interface PreferenceInterface {
+  subject: string;
+  code: string;
+  term: string;
+  able: string;
+  willing: string;
+}
+
+interface PreferenceListInterface {
+  [key: string]: PreferenceInterface;
+}
+
+interface ChildProps {
+  handleCourseChange(course: CourseInterface, value: number): void;
+}
+
+export const SurveyCourseList: React.FC<ChildProps> = (props) => {
+  const { loading, error, data } = useQuery(COURSES, {});
+  const [preferences, setPreferences] = useState<PreferenceListInterface>({});
+
+  useEffect(() => {
+    if (typeof error != "undefined") {
+      console.log(error);
+    }
+  }, [error]);
+
+  useEffect(() => {}, []);
+
+  const calculateRating = (able: string, willing: string) => {
+    if (able === "With Effort" && willing === "Unwilling") {
+      return 20;
+    } else if (able === "Able" && willing === "Unwilling") {
+      return 39;
+    } else if (able === "With Effort" && willing === "Willing") {
+      return 40;
+    } else if (able === "Able" && willing === "Willing") {
+      return 78;
+    } else if (able === "With Effort" && willing === "Very Willing") {
+      return 100;
+    } else if (able === "Able" && willing === "Very Willing") {
+      return 195;
+    } else {
+      return 0;
+    }
+  };
```

Comment:
It's not really a blocker but it's probably a good idea to move some of this data manipulation stuff into it's own file/module

Reply:
moved to new folder called utils/